File: src/main/java/com/rabbitown/yaquest/YaQuest.java
package com.rabbitown.yaquest;

import com.rabbitown.yalib.locale.YLocale;
import com.rabbitown.yaquest.util.Logger;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.HashMap;

/**
 * @author Yoooooory
 */
public class YaQuest extends JavaPlugin {

    public static HashMap<String, HashMap<String, String>> locale = new HashMap<>();

    @Override
    public void onEnable() {
        Logger.prefix = locale.get(new YLocale().getConsoleDefaultLanguage()).get("prefix");
        Logger.info("§aYaQuest v" + getDescription().getVersion() + " enabled!");
    }

    @Override
    public void onDisable() {
        Logger.info("§aYaQuest v" + getDescription().getVersion() + " disabled!");
    }

    @Override
    public void onLoad() {
        loadConfig();
    }

    public void loadConfig() {
        saveDefaultConfig();
        saveResource("language/zh_CN.yml", false);
        YLocale locale = new YLocale();
        YaQuest.locale = locale.getMessageOfHashMap("plugins/YaQuest/language", true);
    }

}

File: src/main/java/com/rabbitown/yaquest/YaQuest.kt
package com.rabbitown.yaquest

import com.rabbitown.yalib.YaLibCentral
import com.rabbitown.yalib.module.locale.I18NPlugin
import com.rabbitown.yalib.module.locale.PrefixLocale
import com.rabbitown.yalib.module.locale.YLocale.Companion.sendLocale
import com.rabbitown.yaquest.command.CommandMain
import com.rabbitown.yaquest.common.Config
import com.rabbitown.yaquest.common.Logger
import com.rabbitown.yaquest.conversation.ConversationManager
import com.rabbitown.yaquest.conversation.internal.ConversationPluginCanceler
import org.bukkit.conversations.ConversationAbandonedEvent
import org.bukkit.entity.Player
import org.bukkit.plugin.java.JavaPlugin

/**
 * @author Yoooooory
 */
class YaQuest : JavaPlugin(), I18NPlugin {

    init {
        instance = this
    }

    override val locale = PrefixLocale(this, "prefix", "zh_CN")

    override fun onLoad() {
        YaLibCentral.registerPlugin(this)
        registerHandlers()
        reload()
    }

    override fun onEnable() {
        Logger.info("我的B站20粉丝啦！")
    }

    override fun onDisable() {
        Logger.info("难道是，天意如此……")
        ConversationManager.runningConversations.removeIf {
            it.abandon(ConversationAbandonedEvent(it, ConversationPluginCanceler(false)))
            (it.forWhom as Player).sendLocale("conversation.cancel-via-disable")
            true
        }
    }

    private fun registerHandlers() {
        CommandMain().register()
    }

    fun reload() {
        locale.load()
        Config.reload()
        ConversationManager.reload()
    }

    companion object {

        lateinit var instance: YaQuest
            private set

    }

}